### PR TITLE
Resolve combine order

### PR DESCRIPTION
**Pull request by gvszg**

### CODE REVIEW

File: app/models/concerns/combinable.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
+module Combinable
+  extend ActiveSupport::Concern
+
+  included do
+    def self.set_order(orders)
+      combined_items_price = orders.map(&:items_price).inject(:+)
+      combined_ship_fee = combined_items_price > 490 ? 0 : 60
```
Comment (varbark) [inline app/models/concerns/combinable.rb]: Cart::FREE_SHIPPING_PRICE = 490
 Cart::SHIP_FEE = 60

File: app/controllers/admin/orders_controller.rb [hunk 2]
```diff
@@ -62,60 +62,19 @@ def select_orders
...
-
-    if errors.present?
-      Rails.logger.error("error: #{errors}")
-      flash.now["danger"] = "請檢查各訂單的明細資料"
-      render :select_orders
-    else
-      target_orders.each { |order| order.update_column(:status, Order.statuses["訂單取消"]) }
-      flash[:notice] = "併單已完成"
-      redirect_to status_index_admin_orders_path
+      order = Order.set_order(combine_orders)
```
Comment (varbark) [inline app/controllers/admin/orders_controller.rb]: set_order 命名不夠一目了然

File: app/controllers/admin/orders_controller.rb [hunk 3]
```diff
@@ -62,60 +62,19 @@ def select_orders
...
+      combine_info = combine_orders.find_by(user_id: order.user_id).info
```
Comment (StevenKo) [inline app/controllers/admin/orders_controller.rb]: combine_order_info vs combine_info vs order_info

File: app/controllers/admin/orders_controller.rb [hunk 4]
```diff
@@ -62,60 +62,19 @@ def select_orders
...
+      combine_items = combine_orders.map(&:items).flatten
```
Comment (StevenKo) [inline app/controllers/admin/orders_controller.rb]: do you need flatten ?

File: app/controllers/admin/orders_controller.rb [hunk 5]
```diff
@@ -62,60 +62,19 @@ def select_orders
...
+      order.set_order_info(combine_info)
+      order.set_order_items(combine_items)
```
Comment (StevenKo) [inline app/controllers/admin/orders_controller.rb]: 不如把combine 的程序都移進 concern

File: app/controllers/admin/orders_controller.rb [hunk 6]
```diff
@@ -62,60 +62,19 @@ def select_orders
   end
 
   def combine_orders
-    target_orders = Order.where(id: params[:selected_order_ids])
-    target_infos = target_orders.map(&:info)
-    target_items = target_orders.map(&:items).flatten
+    combine_orders = Order.where(id: params[:selected_order_ids])
```
Comment (StevenKo) [inline app/controllers/admin/orders_controller.rb]: Order.combime_orders(orders)

File: app/controllers/admin/orders_controller.rb (hunk 4 above)
Comment (gvszg) [inline app/controllers/admin/orders_controller.rb]: 若是拿掉flatten的話，得到的是兩個關聯的集合物件

**Pull request merged**